### PR TITLE
Slot resolution

### PR DESCRIPTION
**Pull request by cdrini**
Todo:
- [ ] Add tests
- [ ] Catch fetch errors + add timeout
- [x] Display error page if slot used up
- [x] Switch to using new default page

### CODE REVIEW

File: src/js/plugins/plugin.search.js [hunk 1]
```diff
@@ -553,10 +553,11 @@ BookReader.prototype.addSearchResult = function(queryString, pageIndex) {
             <span class="pageDisplay">${pageDisplayString}</span>
           </tr>
           <tr>
-            <td>
-              <img class="searchImgPreview" src="${imgPreviewUrl}" />
-            </td>
-            <td>
+            ${page.isViewable ?
+              `<td><img class="searchImgPreview" src="${page.getURI(16, 0)}" /></td>` :
```
Comment (iisa) [inline src/js/plugins/plugin.search.js:557]: what is the significance of 16 & 0?

File: src/js/plugins/plugin.search.js [hunk 2]
```diff
@@ -570,6 +571,30 @@ BookReader.prototype.addSearchResult = function(queryString, pageIndex) {
   }
 };
 
+/**
+ * @private
+ */
```
Comment (iisa) [inline src/js/plugins/plugin.search.js:576]: suggestion: add more context to comment

File: src/js/plugins/plugin.search.js (hunk 1 above)
Comment (cdrini) [reply src/js/plugins/plugin.search.js]: Fixed :+1:

File: src/js/plugins/plugin.search.js (hunk 2 above)
Comment (cdrini) [reply src/js/plugins/plugin.search.js:576]: Added :+1:

**Comment by cdrini** [comment]
Still some jankiness around this specific experience, but good enough for a beta release on IA to get more feedback! Merging. Codecov is apparently out to lunch today.

**Pull request merged**